lib/evss/disability_compensation_form/form4142_processor.rb

```diff
@@ -72,7 +73,11 @@ def submission_date
 
       # Flip this on to use the 2024 PDF template
       def generate_2024_version?
-        Flipper.enabled?(:disability_526_form4142_use_2024_template)
+        @generate_2024_version
+      end
+
+      def determine_2024_version
+        @submission.form[Form526Submission::FORM_4142]['completed2024Form'] == true
```

Comment:
I assume 'completed2024Form' gets added to the form data on the frontend. Is that correct? Where does that occur?

Reply:
Only suggestion would be to extract `@submission.form[Form526Submission::FORM_4142]` into its own method as it's also used on L59.